servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServicetalkHttpServerAttributesGetter.java

```diff
@@ -0,0 +1,89 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry.http;
+
+import io.servicetalk.http.api.HttpRequestMetaData;
+import io.servicetalk.http.api.HttpResponseMetaData;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpServerAttributesGetter;
+
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+final class ServicetalkHttpServerAttributesGetter
+    implements HttpServerAttributesGetter<HttpRequestMetaData, HttpResponseMetaData> {
+
+    static final ServicetalkHttpServerAttributesGetter INSTANCE =
+        new ServicetalkHttpServerAttributesGetter();
+
+    private ServicetalkHttpServerAttributesGetter() {
+    }
+
+    @Nullable
+    @Override
+    public String getUrlScheme(HttpRequestMetaData httpRequestMetaData) {
+        return httpRequestMetaData.scheme();
+    }
+
+    @Override
+    public String getUrlPath(HttpRequestMetaData httpRequestMetaData) {
+        return httpRequestMetaData.path();
+    }
+
+    @Override
+    public String getUrlQuery(HttpRequestMetaData httpRequestMetaData) {
+        return httpRequestMetaData.query();
+    }
+
+    @Override
+    public String getHttpRequestMethod(HttpRequestMetaData httpRequestMetaData) {
+        return httpRequestMetaData.method().name();
+    }
+
+    @Override
+    public List<String> getHttpRequestHeader(HttpRequestMetaData httpRequestMetaData, String name) {
+        CharSequence value = httpRequestMetaData.headers().get(name);
+        if (value != null) {
+            return Collections.singletonList(value.toString());
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public Integer getHttpResponseStatusCode(HttpRequestMetaData httpRequestMetaData,
+                                             HttpResponseMetaData httpResponseMetaData,
+                                             @Nullable Throwable error) {
+        return httpResponseMetaData.status().code();
+    }
+
+    @Override
+    public List<String> getHttpResponseHeader(HttpRequestMetaData httpRequestMetaData,
+                                              HttpResponseMetaData httpResponseMetaData,
+                                              String name) {
+        CharSequence value = httpResponseMetaData.headers().get(name);
+        if (value != null) {
+            return Collections.singletonList(value.toString());
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public String getHttpRoute(HttpRequestMetaData httpRequestMetaData) {
+        return httpRequestMetaData.path();
```

Comment:
Javadoc for `SemanticAttributes.HTTP_ROUTE` says:

> MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it. SHOULD include the application root if there is one.
 
Returning `path()` here is not correct, don't implement this method.

Reply:
true, looking at the path it should contain the path params and that's not possible to find.